basil.js

```diff
@@ -876,14 +876,15 @@ pub.background = function (bgColor) {
   var bgLayerName = "basil.js Background";
   var bgLayer = pub.layer(bgLayerName);
 
-  var lockedState = bgLayer.locked;
   bgLayer.locked = false;
-  pub.arrange(bgLayer, pub.BACK);
+  if (currentDoc().layers[0].name === bgLayerName) {
```

Comment:
I think either you misunderstood me, or I am not getting what you are trying to achieve here. What I meant was that the bgLayer should be moved to the back only if it was newly created, not necessarily always when it is in front. (it could be in front when it already exists as the top layer).

Reply:
Yup that's what I thought was happening here.. only if the layer was JUST created, would it be sent back.. but true.. incase it's for some reason the top layer – then would get unnecessarily forced back. Your approach of checking layer count before making it is better.